docs/knowledgebase/advanced/ss58-address-format.md

```diff
@@ -13,3 +13,97 @@ extensible for this reason.
+```
+
+### Polkadot.js
+
+For verifying an address in your JavaScript projects, you can utilize the functions built
+into the [Polkadot.js API](https://github.com/polkadot-js/api/), which supports ed25519 and
+sr25519 key types.
+
+```javascript
+const { decodeAddress, encodeAddress } = require("@polkadot/keyring");
+const { hexToU8a, isHex } = require("@polkadot/util");
+
+const address = "<addressToTest>";
+
+const isValidAddressPolkadotAddress = () => {
+  try {
+    encodeAddress(isHex(address) ? hexToU8a(address) : decodeAddress(address));
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+const isValid = isValidAddressPolkadotAddress();
+
+console.log(isValid);
+```
+
+### Subscan
+
+You can also manually convert and verify an address is valid by using the web tool on
+[Subscan](https://polkadot.subscan.io/tools/ss58_transform). They have also made the
+[essentials](https://github.com/itering/subscan-essentials/tree/78de8d163a3543a217f0cb3d48c6b9816bb5a231)
+behind their web interface open source so you can take advantage of their
+[Go implemented Utilities](https://github.com/itering/subscan-essentials/tree/78de8d163a3543a217f0cb3d48c6b9816bb5a231/util)
+to suit your needs
+
+### Custom Solution
+
+If you wish to make your own custom solution for validating addresses you will need to consider
+how an address is created. You may find some helpful documentation in the
+[Substrate Github Wiki](<https://github.com/paritytech/substrate/wiki/External-Address-Format-(SS58)>)
+pages, as well as in this [article](https://hackmd.io/@gavwood/r1jTRX2Zr#3-Working-with-SS58-and-account-addresses)
+by Gavin Wood.
+
+##### Python
+
+For some inspiration on developing a solution in Python, you may find the
+[Python Substrate Interface](https://github.com/polkascan/py-substrate-interface),developed by
+Polkascan helpful. Here is where you can find their
+[documentation](https://polkascan.github.io/py-substrate-interface/#substrateinterface.Keypair).
+
+##### Go
+
+Centrifuge has a client [implemented in Go](https://github.com/centrifuge/go-substrate-rpc-client/tree/3e974433f8417e386b033fb64a6ac4971f02c737)
+that has Subkey as a dependency that you could use as a
+[starting point](https://github.com/centrifuge/go-substrate-rpc-client/tree/3e974433f8417e386b033fb64a6ac4971f02c737).
+
+##### C++
+
+To help you along the way with a C++ solution you might want to look at
+[Usetech's API implementation](https://github.com/usetech-llc/polkadot_api_cpp). Here is where they
+[encode and decode addresses](https://github.com/usetech-llc/polkadot_api_cpp/blob/master/src/utils/address.cpp).
+
+##### C#/.Net
+
+Usetech also has a [.Net implementation](https://github.com/usetech-llc/polkadot_api_dotnet)
+which may serve as a reference for dealing with addresses in C#. Here is where you can find
+their [address utilities](https://github.com/usetech-llc/polkadot_api_dotnet/blob/master/Polkadot/src/Utils/Address.cs).
```

Comment:
These sections do not feel super complete/useful to me. I would rather just say something like "please refer to the `[client libraries](link)` for examples of working with the SS58 address format in a number of programming languages" or something.

Reply:
Yeah, I get what you mean. The reason why I included them was that it took me some digging in each of the client libraries just to find the areas where they deal with addresses, I thought it would save folks the initial repo digest by giving them a link to the top-level client library, and then a link into where the address specific code is.

Reply:
Yep, I definitely get where you're coming from. When I'm faced with these types of questions, I always err on the side of completeness and clarity...if the topic isn't specific enough to Substrate development to allow me to dive into it completely and with total clarity, I think it's best to omit it. There is already so much important stuff we need to create and maintain, that I think it's very important to be pretty restrictive with what we choose to document.

Reply:
Trimmed down to repo links to provide some direction